### ISSUE TITLE
Grid focus outline should be redrawn on mousedown

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#6183]
**Originally by @jdahlstrom**

___
When clicking a Grid cell, the focus outline is moved on mouseup instead mousedown, making it look clumsy. Especially if the Grid is initially blurred, the focus cell is initially (on mousedown) drawn on the cell that was focused when Grid last was in focus, and moved to the correct place on mouseup.

___
 Imported from https://dev.vaadin.com/ issue #16484

**Comment by amahdy** [comment]
This is still a valid issue, you dummy stupid bot!

**Comment by amahdy** [comment]
Still an issue.
